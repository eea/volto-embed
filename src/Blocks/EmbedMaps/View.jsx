import React, { useEffect, useMemo } from 'react';
import { connect } from 'react-redux';
import { Message } from 'semantic-ui-react';
import { getContent } from '@plone/volto/actions';
import { flattenToAppURL } from '@plone/volto/helpers';
import EmbedMap from '@eeacms/volto-embed/EmbedMap/EmbedMap';
import { pickMetadata } from '@eeacms/volto-embed/helpers';

function getMaps(props) {
  const content = props.mapsContent || {};
  const maps = content.maps || props.data?.maps || {};
  return {
    ...pickMetadata(content),
    ...maps,
  };
}

function View(props) {
  const { id, getContent, mode } = props;
  const {
    useVisibilitySensor = true,
    with_notes = true,
    with_sources = true,
    with_more_info = true,
    with_share = true,
    with_enlarge = true,
  } = props.data;

  const url = flattenToAppURL(props.data.url || '');

  const maps = useMemo(() => getMaps(props), [props]);

  useEffect(() => {
    const mapsId = maps['@id'] ? flattenToAppURL(maps['@id']) : undefined;
    if (mode === 'edit' && url && url !== mapsId) {
      getContent(url, null, id);
    }
  }, [id, getContent, mode, url, maps]);

  if (props.mode === 'edit' && !url) {
    return <Message>Please select a map from block editor.</Message>;
  }

  if (!maps) {
    return null;
  }

  return (
    <div className="embed-map">
      <EmbedMap
        data={{
          ...maps,
          useVisibilitySensor,
          with_notes,
          with_sources,
          with_more_info,
          with_share,
          with_enlarge,
        }}
        id={props.id}
      />
    </div>
  );
}

export default connect(
  (state, props) => ({
    mapsContent: state.content.subrequests?.[props.id]?.data,
  }),
  { getContent },
)(View);
