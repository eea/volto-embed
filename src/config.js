import ViewIframeBlock from './Iframe/ViewIframe.jsx';
import EditIframeBlock from './Iframe/EditIframe.jsx';

export function applyConfig(config) {
  config.blocks.blocksConfig.maps = {
    ...config.blocks.blocksConfig.maps,
    view: ViewIframeBlock,
    edit: EditIframeBlock,
  };
  return config;
}
