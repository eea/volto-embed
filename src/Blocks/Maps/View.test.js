import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import View from './View';
import installEmbedMaps from '.';

installEmbedMaps(config);

config.settings = {
  ...config.settings,
  publicURL: 'https://www.eea.europa.eu/',
};

config.blocks.blocksConfig = {
  ...config.blocks.blocksConfig,
  maps: {
    id: 'maps',
    title: 'Maps',
    group: 'media',
    extensions: {},
    variations: [],
    restricted: false,
    mostUsed: true,
    sidebarTab: 1,
    security: {
      addPermission: [],
      view: [],
    },
  },
};

describe('Test Maps Block rendering', () => {
  const data = {
    '@type': 'maps',
    url: 'https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166',
    useVisibilitySensor: false,
    dataprotection: {
      enabled: false,
    },
  };

  test('test-1', () => {
    const component = renderer.create(
      <Provider store={global.store}>
        <View data={data} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('test-2', () => {
    const component = renderer.create(
      <Provider store={global.store}>
        <View
          data={{
            ...data,
            height: '100vh',
          }}
        />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('test-3', () => {
    const Component = (props) => (
      <Provider store={global.store}>
        <View
          data={{
            ...data,
            dataprotection: {
              ...data.dataprotection,
              enabled: true,
            },
            height: '100vh',
            useVisibilitySensor: false,
          }}
        />
      </Provider>
    );
    const { container, rerender } = render(<Component />);

    expect(container).toMatchSnapshot();

    container.querySelector('.privacy-button button').click();

    rerender(<Component />);
    expect(container).toMatchSnapshot();
  });
});
