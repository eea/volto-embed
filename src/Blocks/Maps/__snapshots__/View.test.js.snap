// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Maps Block rendering test-1 1`] = `
<div
  className="block maps align center"
>
  <div
    className="maps-inner"
  >
    <div
      className="privacy-protection-wrapper"
      style={
        Object {
          "height": "auto",
          "minHeight": "200px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <iframe
        allowFullScreen={true}
        className="google-map"
        frameBorder="0"
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166"
        style={Object {}}
        title="Embeded ESRI Maps"
      />
    </div>
  </div>
</div>
`;

exports[`Test Maps Block rendering test-2 1`] = `
<div
  className="block maps align center"
>
  <div
    className="maps-inner"
  >
    <div
      className="privacy-protection-wrapper"
      style={
        Object {
          "height": "auto",
          "minHeight": "200px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <iframe
        allowFullScreen={true}
        className="google-map"
        frameBorder="0"
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166"
        style={
          Object {
            "height": "100vh",
          }
        }
        title="Embeded ESRI Maps"
      />
    </div>
  </div>
</div>
`;

exports[`Test Maps Block rendering test-3 1`] = `
<div>
  <div
    class="block maps align center"
  >
    <div
      class="maps-inner"
    >
      <div
        class="privacy-protection-wrapper"
        style="position: relative; overflow: hidden; height: auto; min-height: 200px;"
      >
        <div
          class="privacy-protection"
        >
          <div
            class="overlay"
          >
            <div
              class="wrapped"
            >
              <div
                class="privacy-button"
              >
                <button
                  class="ui primary button"
                >
                  Show external content
                </button>
              </div>
              <div
                class="privacy-toggle"
              >
                <div
                  class="ui checked toggle checkbox"
                >
                  <input
                    checked=""
                    class="hidden"
                    id="remember-choice-undefined"
                    readonly=""
                    tabindex="0"
                    type="checkbox"
                    value=""
                  />
                  <label
                    for="remember-choice-undefined"
                  >
                    Remember my choice
                  </label>
                </div>
              </div>
              <p
                class="discreet"
              >
                Your choice will be saved in a cookie managed by
                 
                .eea.europa.eu
                 that will expire in 
                90
                 days.
              </p>
              <p
                class="discreet"
              >
                <p>
                  <span>
                    This map is hosted by a third party, Environmental Systems Research Institute. By showing the external content you accept the terms and conditions of 
                  </span>
                  <span />
                  <span>
                    . This includes their cookie policies, which we have no control over.
                  </span>
                </p>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Maps Block rendering test-3 2`] = `
<div>
  <div
    class="block maps align center"
  >
    <div
      class="maps-inner"
    >
      <div
        class="privacy-protection-wrapper"
        style="position: relative; overflow: hidden; height: auto; min-height: 200px;"
      >
        <iframe
          allowfullscreen=""
          class="google-map"
          frameborder="0"
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166"
          style="height: 100vh;"
          title="Embeded ESRI Maps"
        />
      </div>
    </div>
  </div>
</div>
`;
